rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php

```diff
@@ -91,13 +91,10 @@ private function shouldSkip(MethodCall $methodCall): bool
             return true;
         }
 
-        if ($this->nodeTypeResolver->isObjectTypes($methodCall->var, [
+        return ! $this->nodeTypeResolver->isObjectTypes($methodCall->var, [
             new ObjectType('League\Event\EventDispatcher'),
             new ObjectType('League\Event\Emitter'),
-        ])) {
-            return false;
-        }
-        return $this->getStaticType($methodCall->args[0]->value) instanceof StringType;
```

Comment:
I am not sure about this. If the value is variable, you have to pass it somehow to anonymous class or not?

Reply:
it needs more fixture to proof that, it was always return true, which the check can be removed. see https://github.com/rectorphp/rector/blob/4b71eaa53d16c9c8cf4a6a2eb1fe4bf6d8006c58/rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php#L96-L100

Reply:
Ok I will create some fixtures with different approaches

Reply:
@samsonasik Failing test added https://github.com/rectorphp/rector/pull/6120
I'll try to add fix